airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3StorageOperations.java

```diff
@@ -130,7 +132,13 @@ public String uploadRecordsToBucket(final SerializableBuffer recordsData,
         exceptionsThrown.add(e);
       }
     }
-    throw new RuntimeException(String.format("Exceptions thrown while uploading records into storage: %s", Strings.join(exceptionsThrown, "\n")));
+    final boolean areAllExceptionsAuthExceptions = exceptionsThrown.stream().filter(e -> e instanceof AmazonS3Exception)
+        .map(s3e -> ((AmazonS3Exception) s3e).getStatusCode()).allMatch(code -> Integer.valueOf(403).equals(code));
```

Comment:
also formatting nit, I'd prefer to have the `allMatch` on a newline to make this easier to read, i.e.
```suggestion
        .map(s3e -> ((AmazonS3Exception) s3e).getStatusCode())
        .allMatch(code -> Integer.valueOf(403).equals(code));
```

(I find it hard to keep track of which parentheses surround which expressions)

Reply:
+1 on the multiline styling, I wasn't sure what the preference here was so I let the autoformatter decide